centreon/src/Core/Host/Infrastructure/Repository/DbReadHostRepository.php

```diff
@@ -745,6 +746,32 @@ public function findByRequestParameters(RequestParametersInterface $requestParam
         return $this->findByRequestParametersAndAccessGroups($requestParameters, []);
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function findByHostGroup(int $hostGroupId): array
+    {
+        $statement = $this->db->prepare($this->translateDbName(
+            <<<'SQL'
+                SELECT host_id, host_name FROM host
+                INNER JOIN hostgroup_relation
+                WHERE hostgroup_hg_id = :hostGroupId
+                AND host.host_register = '1'
```

Comment:
```suggestion
                SELECT host_id, host_name FROM host
                INNER JOIN hostgroup_relation
                ON host_host_id = host_id
                WHERE hostgroup_hg_id = :hostGroupId
                AND host.host_register = '1'
```

Reply:
indeed, I've fixed this on the UpdateHostGroup PR, will add it here